posthog/temporal/data_imports/pipelines/pipeline/utils.py

```diff
@@ -181,13 +181,14 @@ def _evolve_pyarrow_schema(table: pa.Table, delta_schema: deltalake.Schema | Non
             # pyarrow schema to use the larger values so that we're not trying to downscale
             if isinstance(field.type, pa.Decimal128Type) or isinstance(field.type, pa.Decimal256Type):
                 py_arrow_table_column = table.column(field.name)
-                assert isinstance(py_arrow_table_column.type, pa.Decimal128Type) or isinstance(
-                    py_arrow_table_column.type, pa.Decimal256Type
-                )
 
                 if (
-                    field.type.precision > py_arrow_table_column.type.precision
-                    or field.type.scale > py_arrow_table_column.type.scale
+                    isinstance(py_arrow_table_column.type, pa.Decimal128Type)
+                    or isinstance(py_arrow_table_column.type, pa.Decimal256Type)
```

Comment:
Annoyingly `pa.types.is_decimal` doesn't do type guarding for us, which means we can't check the precision and scale

Reply:
Or just in general, how could we fail the assertion before?

Reply:
The next line checks `py_arrow_table_column.type.precision` - this throws an error if `py_arrow_table_column.type` is not of type `Decimal128Type` or `Decimal256Type`. There's no guarantee that the incoming pyarrow table has the same types as those in the delta table which is why we have to check this

Reply:
Ah, I understand. I think my confusion stemmed from not realizing the `isinstance` check at the top was looking at `field.type` but the assertion has `py_arrow_table_column`.

Thank you!